### ISSUE TITLE
Add instrumentation for Kubernetes events

### ISSUE THREAD

**Issue by pavolloffay** [open-telemetry/opentelemetry-operator#565]
Since version 0.38.0 the operator supports injecting instrumentation to workloads via the `Instrumentation.opentelemetry.io` Kind. I would like to extend the CR to add support for generating spans for k8s events. In other words instrumentation for k8s events. 

Kspan: https://github.com/weaveworks-experiments/kspan

![example-2pod](https://user-images.githubusercontent.com/5618424/144408757-220ae811-6eb9-40b2-9e2e-8f2a87a5ecf2.png)

The instrumentation kind could like like this:

```go
type InstrumentationSpec {
  ///existing options
  KubernetesEvents KubernetesEvents
}

type KubernetesEvents {
  Enabled bool
}
```

cc) @bboreham

**Comment by pavolloffay** [comment]
Previously we had a couple of discussions with @jpkrohling about what is the right place for kspan and I more and more think that having it in this operator makes sense.

**Comment by jpkrohling** [comment]
@bboreham, I thought you'd be interested in knowing about this.

**Comment by secustor** [comment]
What would be interesting is to integrate KSpan in the [k8clustereventreceiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/k8seventsreceiver) to allow it to export additionally traces instead of only logs.    
This would also reduce maintenance complexity as we wouldn't have to create a CRD for an external service.

**Comment by pavolloffay** [comment]
That is an excellent point. 

> This would also reduce maintenance complexity as we wouldn't have to create a CRD for an external service.

On the other hand, the CR provides a better user experience. Right now the k8sclustereventreceiver has to be configured with additional k8s objects. We could put this logic to operator - e.g. the operator can recognize the receiver and create the RBAC on behalf of the user.

**Comment by secustor** [comment]
I'm not against creating a CRD, far from it.       
What you are describing, is how I have imagined it.     

This way the user can consume the events outside of the cluster too, e.g. managed k8 cluster provider which want to give consumers "cluster admin" permissions and therefore don't want to setup there tooling inside of the cluster. In case of the CRD this would simply deploy one or more collectors and the necessary rbac controls and point them internal or external OTLP endpoint.

**Comment by pavolloffay** [comment]
Yes, we should definitely consider implementing this as a receiver. Even if we go the receiver route we can still create a specific CRD that will deploy an opinionated version of the collector, or have a field in the existing collector CR to create additional RBAC etc. 


I have other news to share. I am talking to people to figure out how useful it is. So far I am getting positive feedback and it seems that people often correlate k8s events when an incident happens. This capability could be provided as a job that would get events from a given time window (incident window) or just simply run all the time in the cluster if preferred.  

Discusson at k8s instrumentation SIG https://kubernetes.slack.com/archives/C20HH14P7/p1638464478071100. Video recording when kpan was presented there https://youtu.be/Uqyvt9kLnX8?t=1187

**Comment by pavolloffay** [comment]
Another use-case: I have come across a user that could not store events in k8s with a long enough retention. Getting the events to another storage with correlation would solve the issue. I don't know the reasoning why the event store in k8s could not be configured to persist events for long period of time.

**Comment by pavolloffay** [comment]
Crosslinking the issue from the collector-contrib https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/7408
